src/OrchardCore/OrchardCore.DisplayManagement/Notify/NotifyFilter.cs

```diff
@@ -76,10 +80,10 @@ private void OnHandlerExecuting(FilterContext filterContext)
 
         private void OnHandlerExecuted(FilterContext filterContext)
         {
-            var messageEntries = _notifier.List().ToArray();
+            var messageEntries = _notifier.List();
 
             // Don't touch temp data if there's no work to perform.
-            if (messageEntries.Length == 0 && _existingEntries.Length == 0)
+            if (messageEntries.Count == 0 && _existingEntries.Length == 0)
```

Comment:
ToArray() has slightly better performance than ToList()

Reply:
For reference 

https://github.com/OrchardCMS/OrchardCore/discussions/15240#discussioncomment-8348547

Reply:
Not calling something is better than calling it ;) And here we are doing a concat possibly.

Reply:
To rephase, the `Notifier.List()` returns an `IList` we don't need to call `ToArray()` on top of it, it's already materialized, and it has `Count`. then we might call `.Concat()` on it, if it was an array it would create a new one with the concatenation of both. Since internally it's a list, we can assume it can just be better.

Reply:
Yes. I saw the .ToList() on INotifier and thought you used Linq method :).  .Tolist() in this case is a method on the INotifier